File: 第五次練習/1/Rectangle.cpp
#include "Rectangle.h"

Rectangle::Rectangle(double w, double h)
{
	width = w;
	height = h;
}

void Rectangle::setWidth(double newWidth)
{
	width = newWidth;
}

void Rectangle::setHeight(double newHeight)
{
	height = newHeight;
}

double Rectangle::getWidth()
{
	return width;
}

double Rectangle::getHeight()
{
	return height;
}

double Rectangle::getArea()
{
	return width * height;
}

File: python練習/GeometricObject_Inheritance/Rectangle.py
from GeometricObject import GeometricObject

class Rectangle(GeometricObject):
    def __init__(self, width, height, color, filled):
        super().__init__(color, filled)
        self.__width = width
        self.__height = height
        
    def setWidth(self, width):
        self.__width = width
        
    def setHeight(self, height):
        self.__height = height
        
    def getWidth(self):
        return self.__width
    
    def getHeight(self):
        return self.__height
    
    def getPerimeter(self):
        return (int(self.__width) * 2) + (int(self.__height) * 2)
    
    def getArea(self):
        return int(self.__width) * int(self.__height)